Error messages for failure to infer generic parameter and opaque result type should be aligned

**Description**

Consider these two errors:
```swift
func f1<T>() -> T { fatalError() }
f1() // error: generic parameter 'T' could not be inferred

func f2() -> Optional<some Sequence> { 
  return nil // error: cannot infer underlying type for opaque result 'Optional<some P>' from return expression
}
```

Interestingly, the *second error* is a tailored version of the first, because the underlying type of an opaque result type is in fact a generic parameter of the implicit opaque result type declaration. So these errors are fundamentally one and the same, but the difference in their phrasings is hardly enlightening in this respect. A wording for the *second error* that is more similar to that of the much more familiar *first error* would facilitate [our efforts in presenting this feature as an extension of the generics system](https://docs.swift.org/swift-book/LanguageGuide/OpaqueTypes.html#ID614). For example, `underlying type for opaque result type '...' could not be inferred from return expression`.

Sounds Great! So regarding the requirements part, C++ is a daily on the go with high proficiency due to competitive programming, while also having intermediate knowledge in Swift due to iOS Development. Just for my info, skills and credibility of what I know, here are some links (yeah it might not sound much relatable for the current conversation, but maybe you could go through it) : -

[https://github.com/Rajveer100/Personal-CV/blob/main/Resume-Latest.pdf](url) 
[https://www.linkedin.com/in/rajveer-singh-bharadwaj-5504b31a4](url)

Coming back to the "Getting Started" part:

- I have cloned the project successfully via SSH using the commands mentioned.
- Installed dependencies (cmake, ninja, sccache)
- Have done the actual build command which creates this file -> **swift-project/build/Ninja-RelWithDebInfoAssert** (with this error message :-

  **CMake Error at cmake/modules/SwiftSharedCMakeConfig.cmake:16 (find_package):
  Could not find a package configuration file provided by "LLVM" with any of
  the following names:

    LLVMConfig.cmake
    llvm-config.cmake

  Add the installation prefix of "LLVM" to CMAKE_PREFIX_PATH or set
  "LLVM_DIR" to a directory containing one of the above files.  If "LLVM"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  cmake/modules/SwiftSharedCMakeConfig.cmake:253 (swift_common_standalone_build_config_llvm)
  CMakeLists.txt:715 (swift_common_standalone_build_config)**

So what would be the next step after this...

```
/Users/rajveersingh/swift-project/build/Compiler-Diagnostics/GenericParameterError.swift:9:1: error: generic parameter 'T' could not be inferred
f1() // error: generic parameter 'T' could not be inferred
^
/Users/rajveersingh/swift-project/build/Compiler-Diagnostics/GenericParameterError.swift:8:6: note: in call to function 'f1()'
func f1<T>() -> T { fatalError() }
     ^
/Users/rajveersingh/swift-project/build/Compiler-Diagnostics/GenericParameterError.swift:12:10: error: cannot infer underlying type for opaque result 'Optional<some Sequence>' from return expression
  return nil // error: cannot infer underlying type for opaque result 'Optional<some P>' from return expression
         ^
Program ended with exit code: 1
```

--- a/include/swift/AST/DiagnosticsSema.def
+++ b/include/swift/AST/DiagnosticsSema.def
@@ -3918,7 +3918,7 @@ ERROR(unresolved_member_no_inference,none,
 ERROR(cannot_infer_base_of_unresolved_member,none,
       "cannot infer contextual base in reference to member %0", (DeclNameRef))
 ERROR(cannot_infer_underlying_for_opaque_result,none,
-      "cannot infer underlying type for opaque result %0 from return expression",
+      "underlying type for opaque result type %0 could not be inferred from return expression",
       (Type))
 ERROR(unresolved_nil_literal,none,
       "'nil' requires a contextual type", ())

--- a/test/type/opaque.swift
+++ b/test/type/opaque.swift
@@ -581,12 +581,12 @@ do {
 
 // https://github.com/apple/swift/issues/62787
 func f62787() -> Optional<some Collection<Int>> {
-  return nil // expected-error{{cannot infer underlying type for opaque result 'Optional<some Collection<Int>>' from return expression}}
+  return nil // expected-error{{underlying type for opaque result type 'Optional<some Collection<Int>>' could not be inferred from return expression}}
 }
 
 func f62787_1(x: Bool) -> Optional<some Collection<Int>> {
   if x {
-    return nil // expected-error{{cannot infer underlying type for opaque result 'Optional<some Collection<Int>>' from return expression}}
+    return nil // expected-error{{underlying type for opaque result type 'Optional<some Collection<Int>>' could not be inferred from return expression}}
   } 
-  return nil // expected-error{{cannot infer underlying type for opaque result 'Optional<some Collection<Int>>' from return expression}}
+  return nil // expected-error{{underlying type for opaque result type 'Optional<some Collection<Int>>' could not be inferred from return expression}}
 }

--- a/test/type/opaque_return_named.swift
+++ b/test/type/opaque_return_named.swift
@@ -3,11 +3,11 @@
 // Tests for experimental extensions to opaque return type support.
 
 func f0() -> <T> T { return () }
-func f1() -> <T, U, V> T { () } // expected-error{{cannot infer underlying type for opaque result 'T' from return expression}}
+func f1() -> <T, U, V> T { () } // expected-error{{underlying type for opaque result type 'T' could not be inferred from return expression}}
 func f2() -> <T: Collection, U: SignedInteger> T { // expected-note{{required by opaque return type of global function 'f2()'}}
   () // expected-error{{type '()' cannot conform to 'Collection'}}
   // expected-note@-1{{only concrete types such as structs, enums and classes can conform to protocols}}
-  // expected-error@-2{{cannot infer underlying type for opaque result 'T' from return expression}}
+  // expected-error@-2{{underlying type for opaque result type 'T' could not be inferred from return expression}}
 }
 
 func f4() async -> <T> T { () }

